tests/test_validate_call.py

```diff
@@ -803,3 +804,44 @@ def foo(bar: 'list[int | str]') -> 'list[int | str]':
             'input': {'not a str or int'},
         },
     ]
+
+
+@pytest.mark.skipif(sys.version_info < (3, 12), reason='requires Python 3.12+ for PEP 695 syntax with generics')
+def test_validate_call_with_pep_695_syntax() -> None:
+    globs = {}
+    exec(
+        """
+from typing import Iterable
+from pydantic import validate_call
+
+@validate_call
+def max[T](args: Iterable[T]) -> T:
+    return sorted(args, reverse=True)[0]
+        """,
+        globs,
+    )
+    max = globs['max']
+    assert len(max.__type_params__) == 1
```

Comment:
Is there any way to test that `validate_call` is actually using/seeing `T`?

Reply:
Well, I don't think it is: https://github.com/pydantic/pydantic/issues/7796

Which I think is something we can fix separately with the issue above

Reply:
but is there a type param in the pydantic schema or something? or is it replaced with `Any`?

Reply:
It has an arguments schema with an `Any` for the item type

Reply:
In a sense this PR is a bit silly, bc we're fixing the syntax for something that doesn't work, but it at least sets the stage to get the type params in accessible places.